[Pytorch][Executorch] Handwritten view copy out ops should resize out

JacobSzwejbka opened the pull request:
Summary: Handwritten out ops should have feature parity with the codegend ones. This means they should resize out to the appropriate size. Q. Why are these handwritten instead of codegend anyway? Q2. Wheres a good spot to put the resize and copy helpers since they are reused in the codegend out kernels

Test Plan: ci.

Differential Revision: D42177051

larryliu0820 commented on aten/src/ATen/native/TensorShape.cpp:
```diff
@@ -350,6 +350,38 @@ namespace native {
 DEFINE_DISPATCH(stack_serial_stub);
 
+
+namespace {
```
Should we make this reusable by `CompositeViewCopyKernels.cpp`?

JacobSzwejbka replied:
yes, wasnt sure where to put the util functions

larryliu0820 replied:
So I guess it's cleaner to just autogen all these kernels if possible. It seems like most of them are having the same logic as the generated ones.

ezyang replied:
Just put them in a header. If you're not sure what header to put them in, make a fresh one! ViewCopy.h maybe?

larryliu0820 commented on aten/src/ATen/native/TensorShape.cpp:
```diff
@@ -3846,56 +3877,64 @@ at::Tensor lift_fresh(const at::Tensor& self) {
 at::Tensor& _fw_primal_copy_out(const at::Tensor & self, int64_t level, at::Tensor & out) {
   auto tmp = self._fw_primal(level);
-  out.copy_(tmp);
+  resize_out_helper(out, tmp);
```
Actually, I was thinking maybe we should autogen these ops. For example, if you add `autogen` to `_fw_primal_copy` entry [here](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/native_functions.yaml#L13259), I believe the we can delete this kernel.

JacobSzwejbka replied:
yeah I wasnt sure if there was some reason they were hand written, maybe they were added before the codegen thing existed. Ill just change the pr to delete and codegen these instead

JacobSzwejbka commented:
Ok moved most of them to autogen. These two prs confused me on the remaining ones. https://github.com/pytorch/pytorch/pull/89000 and https://github.com/pytorch/pytorch/pull/84579. Basically not sure if this change would break @ezyang's _symint generation, and narrow_copy apparently has some legacy issues I think this might also break?

larryliu0820 commented:
Can you dedup the utils with [`CompositeViewCopyKernels.cpp`](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/templates/CompositeViewCopyKernels.cpp#L21-L61)

larryliu0820 commented:
Also curious if we have test coverage for these ops. Check `test_ops.py` to see if they are covered

JacobSzwejbka commented:
> Can you dedup the utils with [`CompositeViewCopyKernels.cpp`](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/templates/CompositeViewCopyKernels.cpp#L21-L61)

If the symint ops and narrow_copy turn out fine to remove I may not need the utils at all and every op can be moved to codegen.

ezyang reviewed (approved):
Love the deletion here

JacobSzwejbka committed:
[Pytorch][Executorch] Delete handwritten view copy out ops, codegen them instead (#91194)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/91194

Handwritten out ops should have feature parity with the codegend ones. This means they should resize out to the appropriate size. Q. Why are these handwritten instead of codegend anyway? Q2. Wheres a good spot to put the resize and copy helpers since they are reused in the codegend out kernels

Test Plan: ci.

Reviewed By: ezyang, dhruvbird

Differential Revision: D42177051

fbshipit-source-id: 5095d8ed0c64e9bacbc97ff90b60353138611951

JacobSzwejbka commented:
@pytorchbot merge

JacobSzwejbka commented:
@pytorchbot --help

JacobSzwejbka commented:
@pytorchbot --rebase

JacobSzwejbka commented:
@pytorchbot merge -r

JacobSzwejbka commented:
@pytorchbot merge -f "one test failure is due to infra failures and is flaky"
